### PR TITLE
Proper aggregation of istio metrics using prometheus federation

### PR DESCRIPTION
**Pull request by rakesh-garimella**
**Description**

Changes proposed in this pull request:
- Introduce new prometheus instance for scraping istio metrics
- Configure main prometheus to scrape aggregated metrics from federation endpoint
- Unset scrape interval for all servicemonitors in order to have one common default applied
- Drop as much labels as possible from istio metrics

**Related issue(s)**

Fixes: #9701

### CODE REVIEW

**Review by hisarbalik — APPROVED** [review]

**Review by a-thaler — DISMISSED** [review]

**Review by a-thaler — APPROVED** [review]

**Review by piotrmsc — DISMISSED** [review]

**Review by a-thaler — DISMISSED** [review]

File: docs/monitoring/05-04-production-profile.md [hunk 1]
```diff
@@ -27,13 +27,13 @@ The table shows the parameters of each profile and their values:
...
-| **prometheusSpec.volumeClaimTemplate.spec.resources.requests.storage** | Amount of storage requested by the Prometheus Pod. |`10Gi`| `20Gi` | `1Gi` |
-| **prometheusSpec.resources.limits.cpu** | Maximum number of CPUs available for the Prometheus Pod to use. | `600m`| `1` | `150m`|
-| **prometheusSpec.resources.limits.memory** | Maximum amount of memory available for the Prometheus Pod to use. |`2000Mi` | `3Gi` |`800Mi`|
-| **prometheusSpec.resources.requests.cpu** |  Number of CPUs requested by the Prometheus Pod to operate.| `300m`| `300m` | `100m` |
-| **prometheusSpec.resources.requests.memory** | Amount of memory requested by the Prometheus Pod to operate. | `1000Mi`| `1Gi` | `200Mi` |
+| **prometheus.prometheusSpec.retentionSize** | Maximum number of bytes that storage blocks can use. The oldest data will be removed first. | `2GB` | `15GB` | `256MB` |
+| **prometheus.prometheusSpec.retention** | Specifies a period for which Prometheus stores the metrics. |`1d`| `30d` | `2h`|
```
Comment (klaudiagrz) [inline docs/monitoring/05-04-production-profile.md:31]:
```suggestion
| **prometheus.prometheusSpec.retention** | Time period for which Prometheus stores the metrics. |`1d`| `30d` | `2h`|
```

File: docs/monitoring/05-03-prometheus.md [hunk 2]
```diff
@@ -15,5 +15,6 @@ This table lists the configurable parameters, their descriptions, and default va
 
 | Parameter | Description | Default value |
 |-----------|-------------|---------------|
-| **retention** | Specifies a period for which Prometheus stores the metrics in-memory. This retention time applies to in-memory storage only. Prometheus stores the recent data in-memory for the specified amount of time to avoid reading the entire data from disk.| `2h` |
-| **storageSpec.volumeClaimTemplate.spec.resources.requests.storage** | Specifies the size of a Persistent Volume Claim (PVC). | `4Gi` |
+| **prometheusSpec.retention** | Specifies a period for which Prometheus stores the metrics.| `1d` |
+| **prometheusSpec.retentionSize** | Maximum number of bytes that storage blocks can use. The oldest data will be removed first.| `2GB` |
+| **prometheusSpec.storageSpec.volumeClaimTemplate.spec.resources.requests.storage** | Specifies the size of a Persistent Volume Claim (PVC). | `10Gi` |
```
Comment (klaudiagrz) [inline docs/monitoring/05-03-prometheus.md:20]:
```suggestion
| **prometheusSpec.storageSpec.volumeClaimTemplate.spec.resources.requests.storage** | Specifies the size of a PersistentVolumeClaim (PVC). | `10Gi` |
```

File: docs/monitoring/05-03-prometheus.md [hunk 3]
```diff
@@ -15,5 +15,6 @@ This table lists the configurable parameters, their descriptions, and default va
 
 | Parameter | Description | Default value |
 |-----------|-------------|---------------|
-| **retention** | Specifies a period for which Prometheus stores the metrics in-memory. This retention time applies to in-memory storage only. Prometheus stores the recent data in-memory for the specified amount of time to avoid reading the entire data from disk.| `2h` |
-| **storageSpec.volumeClaimTemplate.spec.resources.requests.storage** | Specifies the size of a Persistent Volume Claim (PVC). | `4Gi` |
+| **prometheusSpec.retention** | Specifies a period for which Prometheus stores the metrics.| `1d` |
+| **prometheusSpec.retentionSize** | Maximum number of bytes that storage blocks can use. The oldest data will be removed first.| `2GB` |
```
Comment (klaudiagrz) [inline docs/monitoring/05-03-prometheus.md:19]:
```suggestion
| **prometheusSpec.retentionSize** | Specifies the maximum number of bytes that storage blocks can use. The oldest data will be removed first.| `2GB` |
```

File: docs/monitoring/05-04-production-profile.md [hunk 4]
```diff
@@ -27,13 +27,13 @@ The table shows the parameters of each profile and their values:
...
+| **prometheus.prometheusSpec.storageSpec.volumeClaimTemplate.spec.resources.requests.storage** | Amount of storage requested by the Prometheus Pod. |`10Gi`| `20Gi` | `1Gi` |
+| **prometheus.prometheusSpec.resources.limits.cpu** | Maximum number of CPUs available for the Prometheus Pod to use. | `600m`| `1` | `150m`|
+| **prometheus.prometheusSpec.resources.limits.memory** | Maximum amount of memory available for the Prometheus Pod to use. |`2Gi` | `3Gi` |`800Mi`|
+| **prometheus.prometheusSpec.resources.requests.cpu** |  Number of CPUs requested by the Prometheus Pod to operate.| `200m`| `300m` | `100m` |
+| **prometheus.prometheusSpec.resources.requests.memory** | Amount of memory requested by the Prometheus Pod to operate. | `600Mi`| `1Gi` | `200Mi` |
 | **alertmanager.alertmanagerSpec.retention** | Time period for which Alertmanager retains data.| `120h` | `240h` | `1h` |
 | **grafana.persistence.enabled**| Storing grafana database on a PersistentVolume?|`true`|`true`|`false`|
```
Comment (klaudiagrz) [inline docs/monitoring/05-04-production-profile.md:38]:
```suggestion
| **grafana.persistence.enabled**| Parameter that enables storing Grafana database on a PersistentVolume |`true`|`true`|`false`|
```

File: docs/monitoring/05-03-prometheus.md (hunk 3 above)
Comment (klaudiagrz) [reply docs/monitoring/05-03-prometheus.md:19]: For consistency sake, start either with a verb or with a noun in the context of the whole document.

**Review by a-thaler — APPROVED** [review]

**Review by klaudiagrz — APPROVED** [review]

**Review by piotrmsc — APPROVED** [review]

**Review by m00g3n — APPROVED** [review]

**Review by hisarbalik — APPROVED** [review]

**Pull request merged**